proposals/spi-proposal.md

```diff
@@ -138,90 +152,100 @@ with another SPI device.
+unless `CS_TOGGLE` is also set (in that case toggle between every word).
 
-*When setting `00` (read/write) as the `transferMode`, a word must be written for
-every word received. The user must then extract data from the array of words returned
-in the `SPI_REPLY` message.*
+`0x01: CS_DISABLE` Disable CS pin control (even if a CS pin was specified in the
+`SPI_BEGIN_TRANSACTION` command). Useful to avoid toggling at the end of middle transfers
+in when a large amount of data requires multiple transfers.
 
+`0x02: CS_START_ONLY` Set active state only at beginning of transfer. Useful if breaking
+up a transfer across multiple packets.
 
-#### pinOptions
+`0x03: CS_END_ONLY` Toggle active state only at end of transfer. Useful if breaking up a
+transfer across multiple packets.
 
-|B6   |B5       |B4         |B3           |B2        |B1-B0       |
-|-----|---------|-----------|-------------|----------|------------|
-|0    |CS_TOGGLE|CS_END_ONLY|CS_START_ONLY|CS_DISABLE|transferMode|
 
+### SPI_WRITE
 
-**CS_DISABLE bit (2)** If set, disable CS pin control (even if a CS pin was specified in
-the SPI_BEGIN_TRANSACTION command).
+Only write data, ignoring any data returned by the slave device. No `SPI_REPLY` message
+is sent.
 
-**CS_START_ONLY bit (3)** If set, set active state only at beginning of transfer. Useful
-if breaking up a transfer across multiple packets.
+Provided as a convenience. The same can be accomplished using `SPI_TRANSFRER` and ignoring
+the `SPI_REPLY` message.
 
-**CS_END_ONLY bit (4)** If set, toggle active state only at end of transfer. Useful
-if breaking up a transfer across multiple packets.
+```
+0:  START_SYSEX
+1:  SPI_DATA              (0x68)
+2:  SPI_WRITE             (0x04)
+3:  channel               (HW supports multiple SPI ports. range = 0-3, default = 0)
+4:  csPinControl          (see details in SPI_TRANSFER csPinControl description)
+                            0x00: default behavior (as specified by csPinOptions)
+                            0x01: CS_DISABLE
+                            0x02: CS_START_ONLY
+                            0x03: CS_END_ONLY
+5.  numWords              (number of words to write)
+6:  data 0                (bits 0-6)
+7:  data 0                (bits 7-14 if word size if word size > 7 && < 15)
+8:  data 0                (if word size > 14)
+...                       up to numWords * (wordSize / 7)
+N:  END_SYSEX
+```
+
+### SPI_READ
 
-**CS_TOGGLE bit (5)** If set, toggle CS pin between every word transfered rather than at
-end of transfer.
+Only read data, writing `0` for each word to be read. Reply is sent via `SPI_REPLY`.
 
-*Note: It may be useful to separate pinOptions and transferMode to better scale to
-handle additional use cases as they arise.*
+Provided as a convenience. The same can be accomplished using `SPI_TRANSFRER` and sending
+a `0` for each byte to be read.
 
+```
+0:  START_SYSEX
+1:  SPI_DATA              (0x68)
+2:  SPI_WRITE             (0x05)
```

Comment:
Initially I only had SPI_TRANSFER, but later I added SPI_WRITE and SPI_READ. I think these add some convenience. There is also an ongoing discussion in the Arduino developers email list about about SPI. Adding spiRead and spiWrite functions to the Arduino SPI api is one of the topics being discussed (although they seem to be leaning more towards something like` (spiTransfer(const void *txBuffer, void  *rxBuffer, count)` where readOnly you pass NULL for the txBuffer and writeOnly pass NULL for the rxBuffer.